netdeploy/remote/deployedNetwork.go

```diff
@@ -283,9 +301,302 @@ func (cfg DeployedNetwork) BuildNetworkFromTemplate(buildCfg BuildConfig, rootDi
+			OnCompletion: 0,
+		}
+		tx = transactions.Transaction{
+			Type:                     tp,
+			Header:                   header,
+			ApplicationCallTxnFields: appCallFields,
+		}
+	default:
+		fmt.Printf("%s not supported", tp)
+	}
+
+	t := transactions.SignedTxn{Txn: tx}
+
+	return t, nil
+}
+
+func getGenesisAlloc(name string, allocation []bookkeeping.GenesisAllocation) bookkeeping.GenesisAllocation {
+	for _, alloc := range allocation {
+		if strings.ToLower(alloc.Comment) == strings.ToLower(name) {
+			return alloc
+		}
+	}
+	return bookkeeping.GenesisAllocation{}
+}
+
+func keypair() *crypto.SignatureSecrets {
+	var seed crypto.Seed
+	crypto.RandBytes(seed[:])
+	s := crypto.GenerateSignatureSecrets(seed)
+	return s
+}
+
+func generateInitState(poolAddr basics.Address, sinkAddr basics.Address, genesisID string, genesisHash crypto.Digest, round uint64, accounts map[basics.Address]basics.AccountData, stxn []transactions.SignedTxn) (ledger.InitState, error) {
+
+	var initState ledger.InitState
+	payset := make([]transactions.SignedTxnInBlock, 0, len(stxn))
+	var txibs []transactions.SignedTxnInBlock
+	txibs = make([]transactions.SignedTxnInBlock, 0, len(stxn))
+
+	initBlock := bookkeeping.Block{
+		BlockHeader: bookkeeping.BlockHeader{
+			GenesisID:   genesisID,
+			GenesisHash: genesisHash,
+			Round:       basics.Round(round),
+			RewardsState: bookkeeping.RewardsState{
+				RewardsRate: 1,
+				RewardsPool: poolAddr,
+				FeeSink:     sinkAddr,
+			},
+			UpgradeState: bookkeeping.UpgradeState{
+				CurrentProtocol: protocol.ConsensusCurrentVersion,
+			},
+		},
+	}
+
+	initBlock.RewardsLevel = 0
+
+	for i := 0; i < len(stxn); i++ {
+		txib, err := initBlock.EncodeSignedTxn(stxn[i], transactions.ApplyData{})
+		if err != nil {
+			return ledger.InitState{}, err
+		}
+		txibs = append(txibs, txib)
+	}
+
+	payset = append(payset, txibs...)
+
+	initBlock.Payset = payset
+	var err error
+	initBlock.TxnRoot, err = initBlock.PaysetCommit()
+	if err != nil {
+		return ledger.InitState{}, err
+	}
+
+	initState.Block = initBlock
+	initState.Accounts = accounts
+	initState.GenesisHash = crypto.Hash([]byte(genesisID))
+	return initState, nil
+}
+
+func createBlock(poolAddr basics.Address, sinkAddr basics.Address, genesisID string, genesisHash crypto.Digest, round uint64, prev bookkeeping.Block, stxn []transactions.SignedTxn) (bookkeeping.Block, error) {
+	payset := make([]transactions.SignedTxnInBlock, 0, len(stxn))
+	var txibs []transactions.SignedTxnInBlock
+	txibs = make([]transactions.SignedTxnInBlock, 0, len(stxn))
+
+	block := bookkeeping.Block{
+		BlockHeader: bookkeeping.BlockHeader{
+			GenesisID:   genesisID,
+			GenesisHash: genesisHash,
+			Round:       basics.Round(round),
+			RewardsState: bookkeeping.RewardsState{
+				RewardsRate: 1,
+				RewardsPool: poolAddr,
+				FeeSink:     sinkAddr,
+			},
+			UpgradeState: bookkeeping.UpgradeState{
+				CurrentProtocol: protocol.ConsensusCurrentVersion,
```

Comment:
use the same protocol as the one from the previous block. This is important since we sometimes need to create a simulation of `future` protocol.